### PR TITLE
Update Header to take any JSX.Component for use with UserMenu

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-1827.

#### Changes proposed in this Pull Request:
- refactor Header to take actionItems as HeaderIcon or new HeaderAvatar type
- add "within a Header" story to UserMenu stories

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 ### Changed
 -   Updated styles for the `<Drawer>`.
+-   Updated actionItem prop type for `<Header>`.
```
Comment (joebochill) [inline CHANGELOG.md:12]: I would list this as an 'Added': Ability to use Avatar in `<Header>` actionItems

File: components/src/core/header/headerActionItems.tsx [hunk 2]
```diff
@@ -67,16 +73,29 @@ export const HeaderActionItems: React.FC<ActionItemProps> = (props) => {
...
-                    <TouchableOpacity
-                        key={`action_${index}`}
-                        testID={`header-action-item${index}`}
-                        onPress={actionItem.onPress}
-                        style={[defaultStyles.actionItem, styles.actionItem]}
-                    >
-                        <HeaderIcon IconClass={actionItem.icon} />
-                    </TouchableOpacity>
-                ))}
+                {items.slice(0, MAX_ITEMS).map((actionItem: any, index: number) => {
```
Comment (joebochill) [inline components/src/core/header/headerActionItems.tsx:76]: does actionItem need to be 'any' here? Or can we call it HeaderIconType | HeaderAvatar and cast where needed?

**Review by joebochill — COMMENTED** [review]
Looks good 👍

File: components/src/core/header/headerActionItems.tsx (hunk 2 above)
Comment (daileytj) [reply components/src/core/header/headerActionItems.tsx:76]: Good call...We shouldn't need it now. I missed that in the cleanup.

**Review by joebochill — APPROVED** [review]

**Pull request merged**